draft-ietf-httpbis-digest-headers.md

```diff
@@ -358,152 +438,213 @@ Examples:
+`Digest-algorithm token`. `Digest` also includes an encoded output, the format
+is specified by the value of `Digest-algorithm output format`.
+
+Algorithms used in `Representation-Digest` and `Content-Digest` are communicated
+using the value of `Digest Dictionary key`. The encoding for these fields is
+always sf-binary.
+
+Algorithms used in `Want-Representation-Digest` and `Want-Content-Digest` are
+communicated using the value of `Digest List item`.
 
-The "HTTP Digest Algorithm Values Registry",
-maintained by IANA at <https://www.iana.org/assignments/http-dig-alg/> registers
-digest-algorithm values.
 Registrations MUST include the following fields:
 
- - Digest algorithm: the token value.
-     The registry can be used to reserve token values
+ - Algorithm: a friendly name of the algorithm
+ - Description: a short description of the algorithm
+ - Digest-algorithm token: the token value used in `Digest` and `Want-Digest`
+   fields
+ - Digest-algorithm output format: the encoding format of output used in `Digest`
+ - Digest Dictionary key: the Structured Fields key value used in
+   `Representation-Digest` or `Content-Digest` field Dictionary member keys
+ - Digest List item: the Structured Fields item used as
+   `Want-Representation-Digest` or `Want-Content-Digest` field List members
  - Status: the status of the algorithm.
      Use "standard" for standardized algorithms without known problems;
      "experimental" or some other appropriate value
      - e.g. according to the type and status of the primary document
      in which the algorithm is defined;
      "insecure" when the algorithm is insecure;
      "reserved" when the algorithm references a reserved token value
- - Description: the description of the digest-algorithm and its encoding
- - Reference: a set of pointers to the primary documents defining the digest-algorithm
-
-The associated encoding for new digest-algorithms MUST either
-be represented as a quoted string
-or MUST NOT include ";" or "," in the character sets used for the encoding.
+ - Reference: a set of pointers to the primary documents defining the algorithm
+   and key
 
 Insecure digest algorithms MAY be used to preserve integrity against corruption, but MUST
-NOT be used in a potentially adversarial setting; for example, when signing digest fields' values for
+NOT be used in a potentially adversarial setting; for example, when signing integrity fields' values for
 authenticity.
 
-The registry is initialized with the tokens listed below.
+The registry is initialized with the algorithms listed below.
 
   {: vspace="0"}
-  sha-512
-  : * Digest Algorithm: sha-512
+  SHA-512
+  : * Algorithm: SHA-512
     * Description: The SHA-512 algorithm [RFC6234].
-      The output of this algorithm is encoded using base64 [RFC4648].
+    * Digest-algorithm token: sha-512
+    * Digest-algorithm output format: Base64 [RFC4648]
+    * Digest Dictionary key: sha-512
+    * Digest List item: sha-512
```

Comment:
Not sure we should repeat the same token multiple time, maybe the wg will help in fixing that